node/cn/api.go

```diff
@@ -240,6 +240,49 @@ func (api *PublicDebugAPI) DumpBlock(blockNr rpc.BlockNumber) (state.Dump, error
 	return stateDb.RawDump(), nil
 }
 
+type Trie struct {
+	Type   string `json:"type"`
+	Hash   string `json:"hash"`
+	Parent string `json:"parent"`
+	Path   string `json:"path"`
+}
+
+type DumpStateTrieResult struct {
+	Root  string `json:"root"`
+	Tries []Trie `json:"tries"`
+}
+
+// DumpStateTrie retrieves all state/storage tries of the given state root.
+func (api *PublicDebugAPI) DumpStateTrie(blockNr uint64) (DumpStateTrieResult, error) {
+	//func (api *PublicDebugAPI) DumpStateTrie(blockNr rpc.BlockNumber) error {
```

Comment:
Any reason why there is a comment? Also, `rpc.BlockNumber` might be better idea then `uint64`.

Reply:
I removed.
`rpc.BlockNumber` doesn't support decimal, supports only hex.
So I changed it.